src/Engine/EmitMSIL/Operators.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using ProtoCore.Utils;
+
+namespace EmitMSIL
+{
+    public class Operators
+    {
+        private static string GetOperatorMethodName(string name)
+        {
+            if (name[0] == '%')
+            {
+                name = name.Remove(0, 1);
+            }
+
+            return name[0].ToString().ToUpper() + name.Substring(1);
+        }
+
+        private static bool ToBoolean(dynamic a)
+        {
+            if (a is bool ab)
+            {
+                return ab;
+            }
+
+            if (a is long al)
+            {
+                return al != 0;
+            }
+
+            if (a is null)
+            {
+                return false;
+            }
+
+            if (a is double ad)
+            {
+                return !Double.IsNaN(ad) && !ad.Equals(0.0);
+            }
+
+            if (a is Pointer)
+            {
+                return true;
+            }
+
+            if (a is string str)
+            {
+                return !string.IsNullOrEmpty(str);
+            }
+
+            if (a is char ac)
+            {
+                return ac != 0;
+            }
+
+            return false;
+        }
+
+        public static dynamic Add(dynamic a, dynamic b)
```

Comment:
now that you've implemented all of this as dynamic, it looks pretty clean, I wonder what the performance is vs using object.

This still boxes but I think by using the DLR we get the benefits of cacheing the DLR performs when given different argument types.

Reply:
@sm6srw is the only difference implementation-wise that we save the need for emitting opcodes for boxing and unboxing?

Reply:
I am curious about this too. I will try to read up on the benefits of dynamic vs just using object.

Reply:
Now using objects.